x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials_logic.test.ts

```diff
@@ -0,0 +1,1006 @@
+/*
```

Comment:
This file is long, but it is structured in a very explicit, and predictable pattern.

```
describe('CredentialsLogic', () => {
  describe('actionName', () => {
    describe('tests for valueThatThisActionShouldUpdate1', () => {...}
    describe('tests for valueThatThisActionShouldUpdate2', () => {...}
    describe('tests for valueThatThisActionShouldUpdate3', () => {...}
    it('should not change any other values', () => {...})
```

It's verbose and makes for a long file, but I think it makes it easy to reason about. It also makes it easier to churn through creating tests, as less thought needed to go into how to structure tests for each individual action.

Reply:
I definitely agree with the pattern of grouping our logic tests by actions and what values those actions affect (this should be a pattern that already exists in the codebase).

That being said, I still argue that the length of the file still makes it tougher to reason about than it needs to be, but I would hold the same for any 1000+ line file (IME, any time a file hits 1000+ lines, that's a general code smell that it needs to be simplified, DRYed out, or broken up). I just generally feel strongly that a ~300 line file should not contain a set of tests 3x its size - something can be trimmed down here.

> It also makes it easier to churn through creating tests,

I have to raise a (principled, not specific) objection here (like a red flag to a bull 😆): unit tests should **not** be something we just churn through / go through the motions of. The primary goal of unit tests should be to help developers design robust & loosely coupled software, and serve as living documentation that helps other devs gain a clear overview of the code.

Back to the specific, I actually think your current tests do a really amazing job of that. They really dive into the developer intention of each action/value interaction as well as edge cases and are written/named in an easy-to understand way.

The tests I would like to raise an argument against, however, are the `'should not change any other values'` test. I strongly think we can lose them:

1. Per above, unit tests should **not** be primarily concerned with catching bugs or regressions (far better suited to integration, E2E, and manual testing) - they can **incidentally** do so, but that shouldn't be their driving focus, and clearness / readability of tests should be prioritized foremost.
2. Proving a negative is always a little weird, and usually shouldn't be the focus of an entire test (barring line/branch coverage). In this particular case as well, they're so abstracted that they're difficult to parse, and the multiple `expect.any()`s don't feel like they prove anything.
3. Dropping them trims 200-250+ lines from the file, and loses only 2 lines of branch coverage it looks like (odd that that's even happening, and easy to fix)

I have an alternate suggestion to those tests that should work just as well to check that other values have not changed. Will post it above in your comment specific to a `should not change any other values` test :)

Reply:
I spent a few more hrs reading through the test file and got a medium night's sleep and then woke up and now feel like I was way too full of myself in ☝️ post lol. I definitely now realize that for some tests where a bunch of nested obj values are specifically changing, my proposed solution (checking all values) doesn't actually work as well / will add just as many lines.

I'm still kinda marginal on the `should not change` tests, but if they're helpful for migrating views, I'm good w/ keeping them in - it's not necessarily my preference for writing tests and I'm still a little hand-wavy on it, but definitely not a must-change or blocker to merge for me.

Reply:
6347c61d51a8c97b38bf9d0451f3d88ee22f4de0

Reply:
I'm going to resolve this comment in favor of the other thread we have going about this test file.